### ISSUE TITLE
make-convert composer.lock --> make assumes package versions are in the "webflo" format

### ISSUE THREAD

**Issue by balsama** [drush-ops/drush#2261]
packagist.drupal-composer.org converts standard drupal tags (e.g. 8.x-1.2) to 8.1.2. But the new packager.drupal.org drops the major digit (since it has to packagers, one for 7 and one for 8) and tacks a zero on to the end. E.g. 8.x-1.2 => 1.2.0.

make-convert assumes they are in the "webflo" format:
https://github.com/drush-ops/drush/blob/master/commands/make/make.drush.inc#L674

We currently use this command to generate makefiles for the D.O tarballs, but those makefiles will point to invalid versions of modules once we switch to the nee packager.drupal.org.

I'm not sure if this is something worth fixing since we really shouldn't be spending time supporting the legacy tarball. But I wanted to document it either way.

**Comment by weitzman** [comment]
Thanks for documenting it, and for preferring the Composer way :)

**Comment by greg-1-anderson** [comment]
Yes, this should probably be updated. Distributions are in a bad state right now, as they require drush make; however, drush make is deprecated, as it becomes untenable as soon as a project uses a module with Composer dependencies. This command also helps people transition from make to composer, so we should make sure it supports both formats.

**Comment by balsama** [comment]
Just FYI. We (the developers of lightning) are actually no longer blocked on this. We implemented our own [lock->make converter](https://github.com/acquia/lightning/blob/8.x-2.x/package) because we had issues with how [make-convert pointed to the git repos instead of D.O](https://www.drupal.org/node/2827227).

**Comment by jcnventura** [comment]
We can close this now.
